### DISCUSSION TITLE
Markdown support

### DISCUSSION THREAD

**Discussion by alainter** [streetwriters/notesnook#2747] [Q&A]
Hi,

On the main page of Notesnook, it is mentioned that it has Markdown support.
How can I edit my markdown code (reveal code) or is it possible ?

Is there documentation for using Notesnook, I saw the Notesnook Help site, but I am hoping to find something with a little bit more information. ex: How to use the editor and it's shortcuts.
Or maybe a training video or two

I find Notesnook to be very interesting, but I am having a little bit of difficulty to get my workflow going.
I am used to Joplin, Obsidian, SN and I am finding difficult to find my way with Notesnook .. or maybe it's just not for me.

thanks,

**Comment by Bizzet** [comment]
Notesnook uses a form of visual WYSIWYG (What You See Is What You Get) Markdown rather than separated Markdown.

_Separated Markdown_ is when you edit the file and the software compiles it into the visual form separated from the Markdown source that you edit. This gives you 2 versions of the same file: visual and source. You can usually toggle between the 2 or view the visual version in a split view.

_Visual WYSIWYG Markdown_ is the written the same as the separated markdown except the source is compiled into visual form, in the same editor, as you write it.

You can export Markdown by either:

- Pressing the **Settings** button of each note
- _**Then**_
- - Press **Share** which will allow you to share the Markdown version of your note 
- - _**Or**_
- - Press **Export**
- - Choose **Markdown** which will create a Markdown file for you
- _**Or**_
- Pressing the **Settings** button in the sidebar
- Then pressing **Export All Notes**

If you would like to import Markdown, you can use the **Notesnook Importer**. This allows you to convert Markdown files into the visual Markdown notes that Notesnook uses.

**Sadly, you cannot copy and paste Markdown directly in the editor.** Which is not ideal at all. Not sure why you cannot do this or if it's planned to be added but it is not a feature that is currently available.

If you copy and paste Markdown into the editor, it will look like a regular Markdown file source and nothing will be formatted visually. 

**If you would like to see the visual version of the Markdown, you need to go through every line of Markdown, delete the last character of the Markdown syntax (usually the space right after #), then retype the character.**

Hopefully that eventually gets fixed, because it seems like a relatively easy thing to fix but an extremely poor designed way of doing things from a productivity perspective.

**Comment by alphakode** [comment]
> Notesnook uses a form of visual WYSIWYG (What You See Is What You Get) Markdown rather than separated Markdown.
> 
> _Separated Markdown_ is when you edit the file and the software compiles it into the visual form separated from the Markdown source that you edit. This gives you 2 versions of the same file: visual and source. You can usually toggle between the 2 or view the visual version in a split view.
> 
> _Visual WYSIWYG Markdown_ is the written the same as the separated markdown except the source is compiled into visual form, in the same editor, as you write it.
> 
> You can export Markdown by either:
> 
> - Pressing the **Settings** button of each note
> - _**Then**_
> - - Press **Share** which will allow you to share the Markdown version of your note 
> - - _**Or**_
> - - Press **Export**
> - - Choose **Markdown** which will create a Markdown file for you
> - _**Or**_
> - Pressing the **Settings** button in the sidebar
> - Then pressing **Export All Notes**
> 
> If you would like to import Markdown, you can use the **Notesnook Importer**. This allows you to convert Markdown files into the visual Markdown notes that Notesnook uses.
> 
> **Sadly, you cannot copy and paste Markdown directly in the editor.** Which is not ideal at all. Not sure why you cannot do this or if it's planned to be added but it is not a feature that is currently available.
> 
> If you copy and paste Markdown into the editor, it will look like a regular Markdown file source and nothing will be formatted visually. 
> 
> **If you would like to see the visual version of the Markdown, you need to go through every line of Markdown, delete the last character of the Markdown syntax (usually the space right after #), then retype the character.**
> 
> Hopefully that eventually gets fixed, because it seems like a relatively easy thing to fix but an extremely poor designed way of doing things from a productivity perspective.

It's just nuts! I am used to raw markdown codes edit. WYSIWYG is not for me as I would be importing my existing note and planning on updating them which is nuts in this case. After searching for this specific feature, I have uninstalled it from my machines.

**Comment by tactfuljeer** [comment]
I agree, I need to be able to view the markdown code. In other apps I switch back and forth all the time and for larger files I find the performance is much better viewing the markdown rather than the formatted view.

What’s most important to me though is simply that I occasionally need to copy raw markdown straight out of the app to use somewhere else or share. This is much more efficient than exporting whole notes, opening in another app then copying a section of that note. I really hope it is added soon, great app overall ! There’s just a couple little things I really need.

**Comment by tomtorggler** [comment]
any news on this? it would be great if we could paste markdown and notesnook understood the formatting.

**Reply by n3rdcub** [reply]
This feels insane what exactly is the technical limitation or reasoning behind the current implementation?

It mimics markdown in editor but it's not actual markdown. You can import but then can't edit without using the same methods and shortcuts you used to generate the file in the first place. 

Yet copy pasting isn't supported at all? What in the world....

**Comment by Vague-Scorpio** [comment]
Wondering where Notesnook is with this issue?

Exporting a raw markdown into .md file and then re-inporting the .md file doesn't seem to be working.

I live this app but I might have finally found a fatal flaw. I need copy/paste me arkfown support. Both raw/visual markdown.

It seems like it's an easy fix. Every other notes app has this function. I am not a dev though so maybe it's not as easy as it would seem ..

Any news on this issue?
